crates/pallet-domains/src/block_tree.rs

```diff
@@ -371,13 +375,13 @@ pub(crate) fn process_execution_receipt<T: Config>(
                 let bundle_digests = ExecutionInbox::<T>::get((
                     domain_id,
                     to_prune,
-                    execution_receipt.consensus_block_number,
+                    execution_receipt.consensus_block_number(),
                 ));
                 for (index, bd) in bundle_digests.into_iter().enumerate() {
                     if let Some(bundle_author) = InboxedBundleAuthor::<T>::take(bd.header_hash) {
                         // It is okay to index `ER::bundles` here since `verify_execution_receipt` have checked
                         // the `ER::bundles` have the same length of `ExecutionInbox`
-                        if execution_receipt.inboxed_bundles[index].is_invalid() {
+                        if execution_receipt.inboxed_bundles()[index].is_invalid() {
```

Comment:
Would it be more efficient to set `let inboxed_bundles = execution_receipt.inboxed_bundles()` before the loop rather than repeatedly call the method in a loop?